src/exportVars.ts

```
@@ -0,0 +1,38 @@
+import * as core from '@actions/core'
+import { EnvObject } from './types'
+
+/**
+ * Export key value pair from the EnvObject as environment variables
+ * @param entries The content of the .env file as an EnvObject
+ * @param mask A boolean flag indicating whether to mask the value
+ * @param removeQuotes A boolean flag indicating whether to remove quotes if the value is wrapped in them
+ * @returns {Promise<void>} Resolves with undefined.
+ */
+export async function exportVariables(
+  entries: EnvObject,
+  mask: boolean,
+  removeQuotes: boolean
+): Promise<void> {
+  return new Promise(resolve => {
+    try {
+      for (const [key, value] of Object.entries(entries)) {
+        let finalValue = value
+        if (
+          removeQuotes &&
+          ((value.startsWith('"') && value.endsWith('"')) ||
+            (value.startsWith("'") && value.endsWith("'")))
+        ) {
+          finalValue = value.slice(1, -1)
+        }
+        if (mask) {
+          core.setSecret(finalValue)
+        }
+        core.exportVariable(key, finalValue)
+      }
+      resolve()
+    } catch (error) {
+      if (error instanceof Error)
+        throw new Error(`Error setting environment variables: ${error.message}`)
+    }
```

Should always put an else clause, and throw the original if we don't know what got thrown. Otherwise this thin will silently eat errors.

---

I believe this has been addressed in f77cf96d07e31eaecf2cb45c6fcc4933c2c4c538